### PR TITLE
Create Service part 1

### PR DESCRIPTION
**Pull request by K-Mike**

### CODE REVIEW

File: app/src/main/java/com/streamdata/apps/cryptochat/services/DateService.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import java.util.TimerTask;
+
+public class DateService extends Service{
+    private static Timer timer = new Timer();
+    private static final long REPEAT_TIME = 1000 * 30;
+
+    @Override
+    public int onStartCommand(Intent intent, int flags, int startId) {
+        timer.scheduleAtFixedRate(new Task(), 0, REPEAT_TIME);
+        return START_NOT_STICKY;
```
Comment (cypt) [inline app/src/main/java/com/streamdata/apps/cryptochat/services/DateService.java]: For started services, there are two additional major modes of operation they can decide to run in, depending on the value they return from onStartCommand(): START_STICKY is used for services that are explicitly started and stopped as needed, while START_NOT_STICKY or START_REDELIVER_INTENT are used for services that should only remain running while processing any commands sent to them. See the linked documentation for more detail on the semantics.

Comment (K-Mike) [reply app/src/main/java/com/streamdata/apps/cryptochat/services/DateService.java]: Это намек, что надо использовать  START_STICKY?)

Comment (cypt) [reply app/src/main/java/com/streamdata/apps/cryptochat/services/DateService.java]: Да

File: app/src/main/java/com/streamdata/apps/cryptochat/MainActivity.java [hunk 2]
```diff
@@ -1,18 +1,74 @@
...
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
+
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
-        Intent intent = new Intent(this, MessageListActivity.class);
+        Intent intent = new Intent(this, DateService.class);
+        bindService(intent, serviceConnection, BIND_AUTO_CREATE);
```
Comment (cypt) [inline app/src/main/java/com/streamdata/apps/cryptochat/MainActivity.java:32]: Если ты анбиндишься в onPause то биндится надо в симметричном методе - onResume

File: app/src/main/java/com/streamdata/apps/cryptochat/MainActivity.java [hunk 3]
```diff
@@ -1,18 +1,74 @@
...
 
-        startActivity(intent);
+        super.onPause();
     }
+
+    public void selfDestruct(View view) {
+        TextView textView = (TextView)findViewById(R.id.myTextView);
+
+        if (bound) {
+            textView.setText(myService.getDate());
```
Comment (cypt) [inline app/src/main/java/com/streamdata/apps/cryptochat/MainActivity.java:51]: Что-то не увидел у сервиса метода getDate

File: app/src/main/java/com/streamdata/apps/cryptochat/services/DateService.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    @Override
+    public IBinder onBind(Intent intent) {
+        return null;
+    }
+
+    private class Task extends TimerTask
+    {
+        public void run()
+        {
+            DateFormat dateFormat = new SimpleDateFormat("yyyy/MM/dd HH:mm:ss");
```
Comment (cypt) [inline app/src/main/java/com/streamdata/apps/cryptochat/services/DateService.java:36]: На каком потоке выполняется?

**Pull request closed without merging**